app/javascript/components/rules/activity.js

```diff
@@ -0,0 +1,58 @@
+import React from "react"
+import { AppProvider, Layout, Card, Badge, Caption, TextContainer, Banner } from '@shopify/polaris'
+
+class Activity extends React.Component {
+  render () {
+    return (
+      <AppProvider>
+        <Layout>
+          <Layout.AnnotatedSection
+            id="ruleActivity"
+            title="Activity"
+            description="See the most recent activity pertaining to this rule."
+          >
+            { this.props.events.map((event, i) => {
+              let banner;
+
+              if (event.error) {
+                banner =
+                  <Card.Section>
+                    <Banner status="warning">
+                      <p>There has been a problem executing this rule.</p>
+                    </Banner>
+                  </Card.Section>
+              }
+
+              return(
+                <Card title={ event.name } key={ i }>
+                  { banner }
+
+                  { event.details.map((details, j) => {
+                    let badgeStatus = details.level == "info" ? "info" : "warning";
+
+                    return(
+                      <Card.Section key={ j }>
+                        <TextContainer>
+                          <p>
+                            <Badge status={ badgeStatus }>{ details.level }</Badge>
```

Comment:
I kinda want to identify each `event_detail` with a success/failure kinda thing, but not sure how to materialize it. Idea?

Reply:
Let's think about it. 

It could be that "info" is not required as most cases will be that level.

I do like how it is now: "timestamp/level/message".